wfe/wfe.go

```diff
@@ -464,7 +465,7 @@ func (wfe *WebFrontEndImpl) verifyPOST(ctx context.Context, logEvent *requestEve
 	// Special case: If no registration was found, but regCheck is false, use an
 	// empty registration and the submitted key. The caller is expected to do some
 	// validation on the returned key.
-	if _, ok := err.(core.NoSuchRegistrationError); ok && !regCheck {
+	if _, ok := err.(core.NoSuchRegistrationError); (ok || berrors.Is(err, berrors.NotFound)) && !regCheck {
```

Comment:
I think the reason to have both errors is for deployability, and we'll remove core.NoSuchRegistrationError later. Is that accurate? If so, is there a followup ticket?